cmd/dev_common.go

```diff
@@ -45,20 +45,30 @@ type devCommonConfig struct {
 	dockerOptions   string
 }
 
-func checkDockerRunOptions(options []string) error {
-	//runOptionsTest := "(^((-p)|(--publish)|(--publish-all)|(-P)|(-u)|(--user)|(--name)|(--network)|(-t)|(--tty)|(--rm)|(--entrypoint)|(-v)|(--volume)|(-e)|(--env))((=?$)|(=.*)))"
-	runOptionsTest := "(^((--help)|(-p)|(--publish)|(--publish-all)|(-P)|(-u)|(--user)|(--name)|(--network)|(-t)|(--tty)|(--rm)|(--entrypoint)|(-v)|(--volume))((=?$)|(=.*)))"
-
+func checkDockerRunOptions(options []string, config *RootCommandConfig) error {
+	runOptionsTest := "(^((--help)|(-p)|(--publish)|(--publish-all)|(-P)|(-u)|(--user)|(--name)|(--network)|(-t)|(--tty)|(--rm)|(--entrypoint))((=?$)|(=.*)))"
 	blackListedRunOptionsRegexp := regexp.MustCompile(runOptionsTest)
-	for _, value := range options {
+
+	for ind, value := range options {
 		isInBlackListed := blackListedRunOptionsRegexp.MatchString(value)
 		if isInBlackListed {
 			return errors.Errorf("%s is not allowed in --docker-options", value)
-
+		}
+		if value == "-v" || value == "--volume" {
```

Comment:
Could we be allowing the `--mount` flag also? In the [docker docs](https://docs.docker.com/storage/volumes/#choose-the--v-or---mount-flag#choose-the--v-or---mount-flag) it sounds like a more powerful alternative. Although I suppose since `-v` is what we use everywhere this is fine.

Reply:
Good idea, but `--mount` has a lot more features and will need more robust parsing.. 
```
 --mount 'type=volume,src=<VOLUME-NAME>,dst=<CONTAINER-PATH>,volume-driver=local,volume-opt=type=nfs,volume-opt=device=<nfs-server>:<nfs-path>,"volume-opt=o=addr=<nfs-address>,vers=4,soft,timeo=180,bg,tcp,rw"'
```
I would hold off for now